esp-hal-common/src/soc/esp32s3/psram.rs

```diff
@@ -748,3 +761,1023 @@ pub(crate) mod utils {
+                iomux.gpio[*pin].modify(|_, w| w.fun_drv().variant(3))
+            }
+        }
+    }
+
+    const SPI_MEM_SPICLK_PAD_DRV_CTL_EN: u32 = 1 << 4;
+    const SPI_MEM_SPI_FMEM_SPICLK_FUN_DRV: u32 = 0x00000003;
+    const SPI_MEM_SPI_FMEM_SPICLK_FUN_DRV_S: u32 = 2;
+
+    fn spi_timing_enter_mspi_low_speed_mode(control_spi1: bool) {
+        // Here we are going to slow the SPI1 frequency to 20Mhz, so we need to
+        // set SPI1 din_num and din_mode regs.
+        //
+        // Because SPI0 and SPI1 share the din_num and din_mode regs, so if we
+        // clear SPI1 din_num and din_mode to 0, if the SPI0 flash
+        // module clock is still in high freq, it may not work correctly.
+        //
+        // Therefore, here we need to slow both the SPI0 and SPI1 and related
+        // timing tuning regs to 20Mhz configuration.
+        // Switch SPI1 and SPI0 clock as 20MHz, set its SPIMEM core clock as 80M and set
+        // clock division as 4
+        spi0_timing_config_set_core_clock(SpiTimingConfigCoreClock::SpiTimingConfigCoreClock80m); // SPI0 and SPI1 share the register for core clock. So we only set SPI0 here.
+        spi0_timing_config_set_flash_clock(4);
+        if control_spi1 {
+            // After tuning, won't touch SPI1 again
+            spi1_timing_config_set_flash_clock(4);
+        }
+
+        // Set PSRAM module clock
+        spi0_timing_config_set_psram_clock(4);
+
+        // #if SPI_TIMING_FLASH_NEEDS_TUNING || SPI_TIMING_PSRAM_NEEDS_TUNING
+        //     clear_timing_tuning_regs(control_spi1);
+        // #endif
+    }
+
+    // Set SPI0 FLASH and PSRAM module clock, din_num, din_mode and extra dummy,
+    // according to the configuration got from timing tuning function
+    // (`calculate_best_flash_tuning_config`). iF control_spi1 == 1, will also
+    // update SPI1 timing registers. Should only be set to 1 when do tuning.
+    //
+    // This function should always be called after `spi_timing_flash_tuning` or
+    // `calculate_best_flash_tuning_config`
+    fn spi_timing_enter_mspi_high_speed_mode(control_spi1: bool) {
+        // spi_timing_config_core_clock_t core_clock = get_mspi_core_clock();
+        let core_clock = SpiTimingConfigCoreClock::SpiTimingConfigCoreClock80m;
+
+        let flash_div: u32 = get_flash_clock_divider();
+        let psram_div: u32 = get_psram_clock_divider();
+
+        // Set SPI01 core clock
+        spi0_timing_config_set_core_clock(core_clock); // SPI0 and SPI1 share the register for core clock. So we only set SPI0 here.
+                                                       // Set FLASH module clock
+        spi0_timing_config_set_flash_clock(flash_div);
+        if control_spi1 {
+            spi1_timing_config_set_flash_clock(flash_div);
+        }
+        // Set PSRAM module clock
+        spi0_timing_config_set_psram_clock(psram_div);
+
+        // #if SPI_TIMING_FLASH_NEEDS_TUNING || SPI_TIMING_PSRAM_NEEDS_TUNING
```

Comment:
Can this be removed?

Reply:
it's meant as a reminder for adding that functionality later